app/src/main/kotlin/com/wire/android/util/permission/CaptureVideoRequestFlow.kt

```diff
@@ -8,30 +8,38 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.remember
 import androidx.compose.ui.platform.LocalContext
 
+/**
+ * Flow that will launch the camera to record a video.
+ * This will handle the permissions request in case there is no permission granted for the camera.
+ *
+ * @param onVideoRecorded action that will be executed for Camera's [ActivityResultContract]
+ * @param onPermissionDenied action to be executed when the permissions is denied
+ * @param targetVideoFileUri target file where the media will be stored
+ */
 @Composable
 fun rememberCaptureVideoFlow(
-    shouldPersistUri: (Boolean) -> Unit,
+    onVideoRecorded: (Boolean) -> Unit,
```

Comment:
how about instead of : onVideoRecorded: (Boolean) -> Unit, we pass  : onVideoRecorded: (CameraActivityResult) -> Unit

where CameraActivityResult = CameraActivityResult(val isVideoPersisted : Boolean) 

dunno just thinking out loud because (Boolean) -> Unit does not say much in my opinion but maybe I am overthinking this one :P

Reply:
I think this could stay as it is right now, in case there is some confusion when using this in other possible places, maybe then we can revisit it. Right now, there are only 2 flows that are expected to be used, avatar and sharing an image, but good point anyway 👍

Reply:
True agree with you